.github/workflows/test.yaml

```diff
@@ -99,18 +98,28 @@ jobs:
     - uses: twisted/python-info-action@v1
     - name: Install dependencies
       run: |
-        python -m pip install --upgrade pip tox
+        python -m pip install --upgrade pip tox codecov coveralls
         tox --notest
 
     - name: Test
       run: |
         python --version
         tox -q
 
-    - name: Publish coverage
+    - name: Generate xMl and publish codecov.io
       # We want to publish coverage even on failure.
       if: contains(matrix['tox-env'], 'withcov') || failure()
-      run: tox -e coverage-prepare,codecov-push,coveralls-push
+      run: |
+        python -m coverage xml -o coverage.xml -i
+        python -m codecov -n 'lnx-${{ matrix.python-version }}-${{ matrix.tox-env }}${{ matrix.noipv6 }}' -X search -X gcov -f coverage.xml
+
+    - name: Publish coveralls
+      # We want to publish coverage even on failure.
+      if: contains(matrix['tox-env'], 'withcov') || failure()
```

Comment:
Won't this publish on failure even if `withcov` isn't in the tox env?  I do see this already existed a few lines up.

Reply:
true. Thanks!

Reply:
My point was that this should only publish if `withcov` was specified.  I don't know how to do this in a simple way, and I haven't thought through the not-simple way.  So, just always doing it and ignoring any potential errors when there's no coverage data (which apparently there either aren't or they have already been dealt with, what with the passing CI (or we don't exercise this case)) may be the best that's worth doing anyways.  And that's what we have now.

I guess the tl;dr is that I think `if: always()` is 'wrong' but maybe GHA isn't friendly enough for it to be worth making it right.